transforms/__testfixtures__/arrow-function.output.js

```diff
@@ -23,3 +23,39 @@ var fn6 = () => ({
 });
 
 [1, 2, 3].map(x => x * x);
+
+[1, 2, 3].map(x => x * x);
+
+compare(1, 2, (num1, num2) => num1 > num2);
+
+Promise.resolve()
+.then(function() {
+  console.log('foo');
+}.bind(this, 'a'))
+.then(a => 4);
+
+foo(function(a) {
+  this.bar(function() {
+    return a + this.b;
+  });
+});
+
+foo(function(a) {
```

Comment:
I'm unsure of whether this outer function can actually be converted to an arrow since there is no directly inside it. I don't believe it changes the calling context of the callback to bar. Since I'm not completely sure I went with the safe route and didn't convert that case. I'd be happy to do that if anyone can make a stronger proof of why it should be fine.